### PR TITLE
[Enhancement] Update the compiler option to: -Wall -Wextra -pedantic  -Wpedantic -Werror

### PR DESCRIPTION
**Pull request by Zqy11**
I'm very sorry, I accidentally deleted the last PR record.

### CODE REVIEW

**Comment by Zqy11** [comment]
I don't know why the CI tests failed.

**Comment by Zqy11** [comment]
I don't know why the CI tests failed.

**Comment by Zqy11** [comment]
I can't find the warning code (such 4372) for the warning.
I found that the "zero-size array" warning code of MS C++ is 4200, but it doesn't work for make/gcc in Linux.

I found that gcc has its own format to surpress warnings.
such as :
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wpedantic"
uint8_t cid[0];
#pragma GCC diagnostic pop

It can work in my local environment without "zero-size array" warnings
But I don't know why it failed in the CI tests.

**Comment by Zqy11** [comment]
I have checked the CI agent fail information, but I don't think my code is wrong.
As the error information of CI is different from the code I submitted。

**Comment by er1cthe0ne** [comment]
> I have checked the CI agent fail information, but I don't think my code is wrong.
> As the error information of CI is different from the code I submitted。

Below is the new compile time error because of the new code change from @zhangml merged yesterday, since we raised the compile time error, it will flag all issues on existing code and also any new code merged recently. Please go ahead and fix the below and then we should be good to go.

/var/lib/jenkins/workspace/alcor-control-agent-pr-test/src/comm/aca_grpc.cpp:88:33: error: ISO C++11 requires at least one argument for the "..." in a variadic macro [-Werror]
   ACA_LOG_INFO("Shutdown server");

**Comment by Zqy11** [comment]
OK. I know the reason.
I will update the newest code version daily.


------------------ Original ------------------

File: build/.gitmodules [hunk 1]
```diff
@@ -0,0 +1,56 @@
+[submodule "third_party/zlib"]
```
Comment (er1cthe0ne) [inline build/.gitmodules:1]: be careful on this. Please don't include this addition of submodules in your final change.

File: build/.gitmodules.0 [hunk 2]
```diff
@@ -0,0 +1,51 @@
+[submodule "third_party/zlib"]
```
Comment (er1cthe0ne) [inline build/.gitmodules.0:1]: same here. be careful on this. Please don't include this addition of submodules in your final change.

File: build/aca-machine-init.sh [hunk 3]
```diff
@@ -56,7 +56,9 @@ echo "4--- installing grpc dependencies ---" && \
 # installing grpc and its dependencies
 GRPC_RELEASE_TAG="v1.24.x"
 echo "5--- cloning grpc repo ---" && \
-    git clone -b $GRPC_RELEASE_TAG https://github.com/grpc/grpc /var/local/git/grpc && \
+    git clone -b $GRPC_RELEASE_TAG https://hub.fastgit.org/grpc/grpc /var/local/git/grpc && \
```
Comment (er1cthe0ne) [inline build/aca-machine-init.sh:59]: we will not want to include this change also...

**Comment by Zqy11** [comment]
I am sorry, I just forgot to correct it.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Eric Li"<notifications@github.com&gt;; 
发送时间: 2020年10月23日(星期五) 下午3:02
收件人: "futurewei-cloud/alcor-control-agent"<alcor-control-agent@noreply.github.com&gt;; 
抄送: "张钱宇-中科大计算机"<841971412@qq.com&gt;; "Assign"<assign@noreply.github.com&gt;; 
主题: Re: [futurewei-cloud/alcor-control-agent] Issue 138 (#155)





 
@er1cthe0ne commented on this pull request.
 
 
In build/aca-machine-init.sh:
 &gt; @@ -56,7 +56,9 @@ echo "4--- installing grpc dependencies ---" &amp;&amp; \  # installing grpc and its dependencies  GRPC_RELEASE_TAG="v1.24.x"  echo "5--- cloning grpc repo ---" &amp;&amp; \ -    git clone -b $GRPC_RELEASE_TAG https://github.com/grpc/grpc /var/local/git/grpc &amp;&amp; \ +    git clone -b $GRPC_RELEASE_TAG https://hub.fastgit.org/grpc/grpc /var/local/git/grpc &amp;&amp; \  
we will not want to include this change also...

**Review by er1cthe0ne — APPROVED** [review]
I have removed aca_tests.cpp since it has been replaced by a set of smaller test files.

The change looks good. I am glad all the compile time warnings has been addressed and we are able to raise the warning level up to improve the code quality. Thanks for the contribution.

**Pull request merged**